### PR TITLE
(#183) Ensure models are downloaded before serving in Ollama inference

### LINKED ISSUE #183: ollama inference should verify models are downloaded before serving
Right now, if you start a distribution using `remote::ollama` without the models downloaded, ollama will attempt to download it upon the first inference request:

```
INFO:     Uvicorn running on http://[::]:5001 (Press CTRL+C to quit)
INFO:     ::1:60686 - "POST /inference/chat_completion HTTP/1.1" 200 OK
Pulling model: llama3.1:8b-instruct-fp16
10:16:19.251 [INFO] [chat_completion] HTTP Request: GET http://localhost:11434/api/ps "HTTP/1.1 200 OK"
Generator cancelled
```

However, these models are big – ex. 16GB for the default `llama3.1:8b-instruct-fp16` – so the HTTP request will timeout before the download completes (and abort the download).

We should ensure that the models are downloaded and ready on server startup before serving requests.

### PR DESCRIPTION
**Pull request by frieda-huang**
This commit fixes #183

### CODE REVIEW

File: llama_stack/providers/adapters/inference/ollama/ollama.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 # TODO: Eventually this will move to the llama cli model list command
 # mapping of Model SKUs to ollama models
 OLLAMA_SUPPORTED_SKUS = {
+    "Llama-Guard-3-1B": "xe/llamaguard3:latest",
```
Comment (ashwinb) [inline llama_stack/providers/adapters/inference/ollama/ollama.py:26]: this should be `Llama-Guard-3-8B` right?

File: llama_stack/providers/adapters/inference/ollama/ollama.py [hunk 2]
```diff
@@ -38,6 +40,7 @@ def __init__(self, url: str) -> None:
         self.url = url
         tokenizer = Tokenizer.get_instance()
         self.formatter = ChatFormat(tokenizer)
+        self.model = "Llama3.2-1B-Instruct"
```
Comment (ashwinb) [inline llama_stack/providers/adapters/inference/ollama/ollama.py:43]: not intended?

**Review by ashwinb — COMMENTED** [review]
looks good to me, I have a couple minor comments. @dltn take one look also?

File: llama_stack/providers/adapters/inference/ollama/ollama.py (hunk 1 above)
Comment (frieda-huang) [reply llama_stack/providers/adapters/inference/ollama/ollama.py:26]: Yes. Will fix that.

File: llama_stack/providers/adapters/inference/ollama/ollama.py (hunk 2 above)
Comment (frieda-huang) [reply llama_stack/providers/adapters/inference/ollama/ollama.py:43]: It's intended. Do you think it's good to have a default model?

Comment (ashwinb) [reply llama_stack/providers/adapters/inference/ollama/ollama.py:43]: no we should not have a default model. this is provided by the rest of the infrastructure at runtime. the inference providers don't have any notions of default models.

Comment (frieda-huang) [reply llama_stack/providers/adapters/inference/ollama/ollama.py]: Just made another commit. Lmk if any modification is needed!

File: llama_stack/providers/adapters/inference/ollama/ollama.py [hunk 3]
```diff
@@ -43,10 +45,13 @@ def __init__(self, url: str) -> None:
     def client(self) -> AsyncClient:
         return AsyncClient(host=self.url)
 
-    async def initialize(self) -> None:
+    async def initialize(self, routing_key: str) -> None:
```
Comment (ashwinb) [inline llama_stack/providers/adapters/inference/ollama/ollama.py:48]: no this isn't quite right or needed. when the keys are registered, a `validate_routing_keys()` call is sent to the provider. this results in `self.routing_keys` being set by the time `initialize` is called. so you can just rely on that for this one.

**Review by ashwinb — CHANGES_REQUESTED** [review]
comment inline

File: llama_stack/providers/adapters/inference/ollama/ollama.py (hunk 3 above)
Comment (Ozlm4108) [reply llama_stack/providers/adapters/inference/ollama/ollama.py:48]: - [x] ```

```

Comment (frieda-huang) [reply llama_stack/providers/adapters/inference/ollama/ollama.py]: Got it. Just made some changes.

**Comment by ashwinb** [comment]
Given my comments above I am closing this PR.

**Pull request closed without merging**